docs/rfcs/0007-flow-modules.md

```diff
@@ -0,0 +1,295 @@
+
+### Component Options
+
+
+Given the above example, the `id` of `integrations.mysql "server1"` would be `module.single.mysql.integrations.mysql.server1`. The `data-agent` field would also be prefixed. There are some inherent issues, deeply nested metrics are likely to run into Prometheus label value limits. On Windows platforms there could be issues with the `data-agent` length. These are issues that currently exist in Agent Flow but are more easily hit using deeply nested modules.
+
+
+### Failure Modes
+
+#### Option 1: When a module fails then fail itself and children.
+
+If an error occurs while re-evaluating a module then the module marks itself as unhealthy and unloads the original module and all children.
+
+*Pros*
+
+* Simple to implement
+* Easy to understand
+
+*Cons*
+
+* One failure mode can cascade
+
+#### Option 2: Modules keep last good value
+
+If an error occurs while re-evaluating a module then the module marks itself as unhealthy and attempts to keep the original module. This may have an issue with cascading failures, if a module depends on a module then the system may enter an inconsistent state while applying and then rolling back the change.
+
+For example, `Module A` has two sub-modules `Module B` and `Module C`. During reevaluation `Module B` reloads appropriately but `Module C` fails. `Module A` unloads both modules and then reloads the last good string. In the case that the last good string also fails then `Module A` is unhealthy and non-functional and `Module A's` submodules do not exist.
+
+*Pros*
+
+* Allows more resilient usage
+
+*Cons*
+
+* Can create undefined behavior
+* Complex to unload and reload
+
+## `modules.multiple`: Allowing multiple modules to be loaded at once
+
+Note: This feels the most experimental of the topics listed.
+
+Exports are accessed via `module.multiple.LABEL.exports.NAME`, `exports` is a `map(array)`. The `exports` label is used to prevent any collision if other fields are added to `modules.multiple`
+
+## Option 1: No filter 
+
+Depend on filtering the `map(string)` input before loading the modules.
+
+
+#### Filter prior
+
+Assume all files in `/configs/*river` are appropriate for the `module.multiple.load`. This has the advantage of being simplifying the usage of modules and development. This also mirrors `module.single` in the being simple and pushing verisoning/usage to other components. Downside is users are unable to put all configs in one folder and magically work.
+
+```river
+
+// Note this doesnt exist and should only be used for representative purposes.
+local.files "loadfolder" {
+    folder_path = "/configs" # Assume this outputs a map(string)
+    filter = "*.river"
+}
+
+module.multiple "load" {
+    source = local.files.loadfolder.contents
+}
+
+prometheus.scrape "module" {
+    // The module.multiple coalesces multiple exports into an array. 
+    targets = module.multiple.load.exports.targets
+}
+
+```
+
+## Option 2: Filter
```

Comment:
A third option would be to say that the set of module sources gets combined into a single module, such that arguments, exports, and component names must be globally unique. 

I'm not sure I understand what the value in having multiple distinct module sources being loaded as multiple modules from one component is yet, so it feels to me like combining the sources into a single module is equivalent in terms of convenience.